scripts/src/javatests/org/oppia/android/scripts/todo/TodoCheckTest.kt

```diff
@@ -0,0 +1,422 @@
+    val testContent1 =
+      """
+      // TODO(#11002): test summary 1.
+      #TODO(#11004): test summary 2.
+      test Todo
+      test TODO
+      """.trimIndent()
+    val testContent2 =
+      """
+      // TODO(#11001): test summary 3.
+      todo
+      <!-- TODO(#11003): test summary 4-->
+
+      """.trimIndent()
+    tempFile1.writeText(testContent1)
+    tempFile2.writeText(testContent2)
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(TODO_CHECK_PASSED_OUTPUT_INDICATOR)
+  }
+
+  @Test
+  fun testTodoCheck_onlyPoorlyFormattedTodosPresent_checkShouldFail() {
+    val testJSONContent =
+      """
+      []
+      """.trimIndent()
+    val testJSONFile = tempFolder.newFile("testfiles/open_issues.json")
+    testJSONFile.writeText(testJSONContent)
+    val tempFile = tempFolder.newFile("testfiles/TempFile.txt")
+    val testContent =
+      """
+      // TODO (#1044): test
+      # TODO(102)
+      <!-- TODO(#   101)-->
+      // some test conent TODO(#1020000): test description.
+      """.trimIndent()
+    tempFile.writeText(testContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(TODO_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      TODOs not in correct format:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:2
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:3
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:4
+      The TODO should be in the format: TODO(#ISSUE_NUMBER): <todo_description>
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testTodoCheck_onlyOpenIssueFailureTodosPresent_checkShouldFail() {
+    val testJSONContent =
+      """
+      [{"number":10000000},{"number":100000004}]
+      """.trimIndent()
+    val testJSONFile = tempFolder.newFile("testfiles/open_issues.json")
+    testJSONFile.writeText(testJSONContent)
+    val tempFile = tempFolder.newFile("testfiles/TempFile.txt")
+    val testContent =
+      """
+      // TODO(#104444444): test summary 1.
+      #TODO(#10210110): test summary 2.
+      test todo
+      some test content Todo
+      <!-- TODO(#101000000): test summary 3-->
+      """.trimIndent()
+    tempFile.writeText(testContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(TODO_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      TODOs not corresponding to an open issue:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:2
+      - ${retrieveTestFilesDirectoryPath()}/TempFile.txt:5
+      Note that, every TODO must correspond to an open issue on GitHub
```

Comment:
Maybe to simplify the error message:

```suggestion
      Every TODO must correspond to an open issue on GitHub
```

Reply:
Done.